dsc_lib/src/dscresources/command_resource.rs

```diff
@@ -282,13 +282,31 @@ pub fn invoke_test(resource: &ResourceManifest, cwd: &str, expected: &str) -> Re
                     return Err(DscError::Operation(format!("Failed to parse json from test {}|{}|{} -> {err}", &test.executable, stdout, stderr)))
                 }
             };
-            let diff_properties = get_diff(&expected_value, &actual_value);
-            Ok(TestResult::Resource(ResourceTestResponse {
-                desired_state: expected_value,
-                actual_state: actual_value,
-                in_desired_state: diff_properties.is_empty(),
-                diff_properties,
-            }))
+
+            // Special case for PSAdapters - Test operation in them returns a single 'InDesiredState' property
+            if (resource.resource_type == "Microsoft.DSC/PowerShell")
```

Comment:
I don't like special casing resources.  I would prefer we wait until resources can propagate metadata and passes up a `inDesiredState` boolean which would be used universally

Reply:
ok; As discusses this will be implemented after #467 using metadata.